### PR TITLE
feat: add thumbnails to research lists

### LINKED ISSUE #2682: [feature request] add thumbnail images to the research listing page
To include picture and better indication of what the 3 icons / numbers mean for each research item.

### PR DESCRIPTION
**Pull request by goratt12**
PR Checklist

- [ ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Added thumbnails to research list

## Git Issues

Closes #2682

## Screenshots/Videos

image

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by AlfonsoGhislieri** [comment]
Thanks for this @goratt12, this makes me think we need to have a think about how to actually move forward in regards to generating and serving thumbnails. The current option that is proposed in this PR I think poses some issues for the resources that are needed and the sacalability moving forwards. To avoid having to pay off this debt down the line, I think we  should re-think the implementation.

Discussion on the topic in: 
https://discord.com/channels/586676777334865928/1148720148509900862

The two options that stand out to me:

- Storing the thumbnail in the DB (adding it to the items on upload and edit to avoid them going stale) --> this does require us to migrate all old howtos
- Using a cdn to service these images --> I wonder if we do have enough traffic/images to really need this approach, might be overengineering at this point?

I think this is definitely an area that could use some discussion before going ahead with implementing thumbnails, @ONEARMY/maintainers.

**Comment by davehakkens** [comment]
Putting this here for visibility. Its an issue that had been opened up before https://github.com/ONEARMY/community-platform/issues/2548

No strong opinion on CDN vs resizing. But agree good moment to tackle the image-size issue side wide!

**Comment by davehakkens** [comment]
hi @goratt12 meanwhile we figure out the discussion on the images. Would you be up for making it mobile ready? A bit bumpy atm
afbeelding

**Comment by goratt12** [comment]
Yeah sure, I’ll have a look at it @davehakkens

**Comment by goratt12** [comment]
Now the thumbnail will be hidden on mobile.

File: src/assets/images/default-research-thumbnail.jpg
Comment (thisislawatts) [inline src/assets/images/default-research-thumbnail.jpg:1]: @goratt12 As this image is only being used for a block that is ~100x100px I've downscaled the image to be 200x200px. This reduces the image size from 1.31mb to 11.7 kb

File: src/pages/Research/Content/ResearchListItem.tsx [hunk 1]
```diff
@@ -180,6 +196,18 @@ const ResearchListItem = ({ item }: IProps) => {
   )
 }
 
+const getItemThumbnail = (researchItem: IResearch.ItemDB): string => {
+  if (researchItem.updates?.length) {
+    const latestImage = researchItem?.updates
```
Comment (thisislawatts) [inline src/pages/Research/Content/ResearchListItem.tsx:201]: **issue(blocking):** Please note that we should only load images from Updates which are publicly available. It is possible for an author to mark an update as either draft or deleted. In this case the author would not expect the image to be visible to the public.

Here's the logic [available elsewhere](https://github.com/ONEARMY/community-platform/blob/cd6ef960b50f64e59d2c3b50935c0a7e43b45e4d/src/pages/Research/Content/ResearchArticle.tsx#L254C1-L256C2):
```ts
const isUpdateVisible = (update: IResearch.UpdateDB) => {
  return update.status !== 'draft' && !update._deleted
}
```

It is being used in the same file to [filter updates for display](https://github.com/ONEARMY/community-platform/blob/cd6ef960b50f64e59d2c3b50935c0a7e43b45e4d/src/pages/Research/Content/ResearchArticle.tsx#L193C1-L195C40). It is probably worth introducing a shared helper for fetching these updates, perhaps something like `getPublicUpdates(researchItem.updates)`

Comment (thisislawatts) [inline src/pages/Research/Content/ResearchListItem.tsx:201]: **issue(blocking):** Please note that we should only load images from Updates which are publicly available. It is possible for an author to mark an update as either draft or deleted. In this case the author would not expect the image to be visible to the public.

Here's the logic [available elsewhere](https://github.com/ONEARMY/community-platform/blob/cd6ef960b50f64e59d2c3b50935c0a7e43b45e4d/src/pages/Research/Content/ResearchArticle.tsx#L254C1-L256C2):
```ts
const isUpdateVisible = (update: IResearch.UpdateDB) => {
  return update.status !== 'draft' && !update._deleted
}
```

It is being used in the same file to [filter updates for display](https://github.com/ONEARMY/community-platform/blob/cd6ef960b50f64e59d2c3b50935c0a7e43b45e4d/src/pages/Research/Content/ResearchArticle.tsx#L193C1-L195C40). It is probably worth introducing a shared helper for fetching these updates, perhaps something like `getPublicUpdates(researchItem.updates)`

**Review by thisislawatts — CHANGES_REQUESTED** [review]

File: src/pages/Research/Content/ResearchListItem.tsx (hunk 1 above)
Comment (goratt12) [reply src/pages/Research/Content/ResearchListItem.tsx]: I created a shared function in the `src\utils\helpers.ts` file named `getPublicUpdates`.
linked both the thumbnail code and the `ResearchArticle.tsx` code

**Review by thisislawatts — APPROVED** [review]

**Pull request merged**